### PR TITLE
Add script for enriching members and organizations of a tenant

### PR DESCRIPTION
**Pull request by yeganathan18**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at 1c425c4</samp>

Added an option to enable or disable frontend notifications for bulk member enrichment tasks. This affects the `sendBulkEnrichMessage` function in `backend/src/serverless/utils/nodeWorkerSQS.ts` and the `bulkEnrich` function in `backend/src/services/premium/enrichment/memberEnrichmentService.ts`. The option is passed as a parameter in the SQS message payload.
​

### <samp>🤖 Generated by Copilot at 1c425c4</samp>

> _Sing, O Muse, of the cunning code that was written_
> _By the skilled developers, who sought to enrich the members_
> _Of their platform, with data from various sources_
> _And to give them the choice, to receive or not the notices_

### <samp>🤖 Generated by Copilot at 1c425c4</samp>

*  Add optional `notifyFrontend` parameter to control websocket messages for bulk enrichment process ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1107/files?diff=unified&w=0#diff-20f166801e8067a398e9206de5dd17d9b76422cf500863bdd97026d339c50687L135-R139), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1107/files?diff=unified&w=0#diff-20f166801e8067a398e9206de5dd17d9b76422cf500863bdd97026d339c50687R145), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1107/files?diff=unified&w=0#diff-b83106cc7e05b9c0cbde32d3e0359c83630c0bb0a2a0257d97ee2a7bbf89f807L141-R141), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1107/files?diff=unified&w=0#diff-b83106cc7e05b9c0cbde32d3e0359c83630c0bb0a2a0257d97ee2a7bbf89f807L173-R206))

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screehshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [x] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: backend/src/bin/scripts/enrich-members-organizations.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          const totalOrganizations = organizations.count
+
+          const enrichmentService = new OrganizationEnrichmentService({
+            options: userContext,
+            apiKey: ORGANIZATION_ENRICHMENT_CONFIG.apiKey,
+            tenantId,
+            limit: totalOrganizations,
+          })
+
+          const enrichedOrgs = await enrichmentService.enrichOrganizationsAndSignalDone()
```
Comment (epipav) [inline backend/src/bin/scripts/enrich-members-organizations.ts:122]: let's also process async here using queue messages - an example is in `organizationEnricher.ts`

File: backend/src/bin/scripts/enrich-members-organizations.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    alias: 't',
+    type: String,
+    description: 'The unique ID of tenant that you would like to enrich.',
+  },
+  {
+    name: 'help',
+    alias: 'h',
+    type: Boolean,
+    description: 'Print this usage guide.',
+  },
```
Comment (epipav) [inline backend/src/bin/scripts/enrich-members-organizations.ts:32]: let's also add two flags -o and -m so in the future we can enrich tenants selectively (ie: only enrich members or only enrich organizations) When both flags are present, or no flags given at all we can enrich both (like the default case)

File: backend/src/bin/scripts/enrich-members-organizations.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    header: 'Options',
+    optionList: options,
+  },
+]
+
+const usage = commandLineUsage(sections)
+const parameters = commandLineArgs(options)
+
+if (parameters.help || !parameters.tenant || !parameters.plan) {
+  console.log(usage)
```
Comment (epipav) [inline backend/src/bin/scripts/enrich-members-organizations.ts:53]: parameters.plan doesn't exist as an option

**Review by epipav — CHANGES_REQUESTED** [review]
looking good :+1:  I added few comments

**Review by epipav — COMMENTED** [review]
lgtm :+1:  included one nitpick
also, can u try the script out on staging and check normal enriching tru UI still works with credits and toast messages

File: backend/src/bin/scripts/enrich-members-and-organizations.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    optionList: options,
+  },
+]
+
+const usage = commandLineUsage(sections)
+const parameters = commandLineArgs(options)
+
+if (parameters.help || !parameters.tenant || !parameters.organization || !parameters.member) {
+  console.log(usage)
+} else {
```
[the remark this answers is not in the document]
Comment (yeganathan18) [reply backend/src/bin/scripts/enrich-members-and-organizations.ts:67]: Got it! thanks for the comment, resolved it 😄

**Review by epipav — APPROVED** [review]

**Pull request merged**